### PR TITLE
Improve JavaScript behavior at coordinator startup

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Improve JavaScript behavior at coordinator startup
Forward port of https://github.com/arangodb/arangodb/pull/12770
Documentation PR: https://github.com/arangodb/docs/pull/563
Oskar PR: https://github.com/arangodb/oskar/pull/270

- [x] :hankey: Bugfix 
- [x] :pizza: New feature 
- [x] :hammer: Refactoring 
- [x] :book: CHANGELOG entry made
- [x] :muscle: The behavior in this PR was *manually tested*
- [ ] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [x] Backports required for: 3.6

### Testing & Verification

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12114/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12321/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12331/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12344/

**Comment by jsteemann** [comment]
Tests blue

File: js/client/modules/@arangodb/testsuites/restart.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  let testCode;
+  // \n's in testCode are required because of content could contain '//' at the very EOF
+  if (file.indexOf('-spec') === -1) {
+    let testCase = JSON.stringify(options.testCase);
+    if (options.testCase === undefined) {
+      testCase = '"undefined"';
+    }
+    testCode = 'const runTest = require("jsunity").runTest;\n ' +
+      'return runTest(' + JSON.stringify(file) + ', true, ' + testCase + ');\n';
+  } else {
```
Comment (dothebart) [inline js/client/modules/@arangodb/testsuites/restart.js:77]: you probably don't need this else?

Comment (jsteemann) [reply js/client/modules/@arangodb/testsuites/restart.js:77]: It is there to support `-spec` tests at some point, if somebody feels the desire to write some.

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12353/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12377/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12392/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12404/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12419/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12434/

**Comment by jsteemann** [comment]
Tests blue

**Review by mchacki — APPROVED** [review]
LGTM

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13217/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13220/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**